Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -632,25 +632,38 @@ function Add-AzureSqlDatabaseServerFirewallRule
           [String] [Parameter(Mandatory = $true)] $firewallRuleName)
     
     Trace-VstsEnteringInvocation $MyInvocation
-    Write-Verbose "Creating firewall rule $firewallRuleName"
 
-    $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
-
-    if(IsLegacyAzureConnection $connectionType)
-    {
-        Add-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
-    }
-    elseif (IsAzureRmConnection $connectionType)
+    try
     {
+        Write-Verbose "Creating firewall rule $firewallRuleName"
+
+        $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
 
-        Add-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        if(IsLegacyAzureConnection $connectionType)
+        {
+            Add-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        }
+        elseif (IsAzureRmConnection $connectionType)
+        {
+
+            Add-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        }
+        else
+        {
+            throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        }
+
+        Write-Verbose "Firewall rule $firewallRuleName created"
     }
-    else
+    catch 
     {
-        throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        $parsedException = Parse-Exception($_.Exception)
```

Comment:
No that's not completely true , sometime $response stream comes out to be null , in that case method parse-exception will return true.
I am beveling $_.Exception will never be null because it have reached in try catch because it got some exception.